### PR TITLE
MM-18911: Filtering search by channel should also show the channel name and not only its ID

### PR DESCRIPTION
**Pull request by allenlai18**
#### Summary

Web app filter search `in:` to display channel name (channel id) and display name like mobile

https://user-images.githubusercontent.com/7275313/68829461-d300ed80-065d-11ea-9642-2fbfdadbf6a2.PNG

#### Ticket Link

( https://mattermost.atlassian.net/browse/MM-18911 )

### CODE REVIEW

File: components/suggestion/search_channel_provider.jsx [hunk 1]
```diff
@@ -21,6 +21,9 @@ function itemToName(item) {
     if (item.type === Constants.GM_CHANNEL) {
         return '@' + item.display_name.replace(/ /g, '');
     }
+    if (item.type === Constants.OPEN_CHANNEL || item.type === Constants.PRIVATE_CHANNEL) {
+        return item.display_name + ' ~(' + item.name + ')';
```
Comment (Willyfrog) [inline components/suggestion/search_channel_provider.jsx]: the tilde character `~` should be close to the channel name, as it is mattermost indicator that it is a channel. If you look at the picture you added for mobile it is like that. Still, I'd like @esethna to confirm that this is the way we want to show to the user

Comment (allenlai18) [reply components/suggestion/search_channel_provider.jsx]: > this should only change the suggestion for channel autocompletion, so the user can identify a channel by either its id or its name, depending on what's familiar for him. With the proposed change it is adding extra info to the search and nothing can be found.
> 
> Let me know if you need any more details on the approach

are you saying this changes things besides the channel autocompletion and its breaking other things? it's not finding anything?

**Comment by allenlai18** [comment]
> 1. The test server shows a different character, it should be the tilde as displayed here: ~
> 2. The tilde should be within the parentheses (~some-channel)
> 
> image
> 
> Other than that looks good, adding Katie as this touches her area

Moved the tilde inside the parentheses.  @esethna   do you think it looks like a '-' because of the browser?  it looks like a tilde on mine.

**Comment by Willyfrog** [comment]
@allenlai18 this prevents searching, it should display as you did, but search functionality shouldn't break because of this change.

**Comment by allenlai18** [comment]
> @allenlai18 this prevents searching, it should display as you did, but search functionality shouldn't break because of this change.



> @allenlai18 this prevents searching, it should display as you did, but search functionality shouldn't break because of this change.

hmmm. What part of searching is breaking? It seems to be working for me

**Comment by wiersgallak** [comment]
I am seeing a tilde in my testing on Mac and it is correctly placed inside of the parentheses. 
Screen Shot 2019-11-15 at 8 06 46 AM
I have added another channel "Katie Test (~apple).  The channel name is not being identified by autocomplete  in a search starting with "t" 
Screen Shot 2019-11-15 at 8 20 41 AM

Screen Shot 2019-11-15 at 8 20 30 AM

**Comment by allenlai18** [comment]
> I am seeing a tilde in my testing on Mac and it is correctly placed inside of the parentheses.
> Screen Shot 2019-11-15 at 8 06 46 AM
> I have added another channel "Katie Test (~apple). The channel name is not being identified by autocomplete in a search starting with "t"
> Screen Shot 2019-11-15 at 8 20 41 AM
> 
> Screen Shot 2019-11-15 at 8 20 30 AM

I think that is the expected behavior. katieTest is camel cased. if it was seperated with dashes it works.  what @Willyfrog was talking about is the search is broken when you search for posts within a channel.

**Comment by allenlai18** [comment]
> @allenlai18 this prevents searching, it should display as you did, but search functionality shouldn't break because of this change.

Fixed! Let me know what you think.

**Comment by allenlai18** [comment]
@Willyfrog   can you merge when you get a chance?

**Pull request merged**